### PR TITLE
more utility functions and better docs for nested tuples

### PR DESCRIPTION
**Pull request by jdegoes**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -281,6 +353,14 @@ tuple2 :: forall a b z. (a -> b -> z) -> Tuple2 a b -> z
 
 Given a function of 2 arguments, return a function that accepts a 2-tuple.
 
+#### `unTuple2`
+
+``` purescript
+unTuple2 :: forall a b z. (Tuple2 a b -> z) -> a -> b -> z
```
Comment (paf31) [inline README.md]: `curry2`?

File: README.md [hunk 2]
```diff
@@ -273,6 +273,78 @@ type Tuple10 a b c d e f g h i z = Tuple (Tuple9 a b c d e f g h i) z
 ```
 
 
+#### `entuple2`
+
+``` purescript
+entuple2 :: forall a b. a -> b -> Tuple2 a b
```
Comment (paf31) [inline README.md]: While we're breaking stuff, I think I'd rather call these `tupleN`, and the functions `(un)curryN`.

**Comment by jdegoes** [comment]
Done.

**Comment by paf31** [comment]
Wow, quick :) I meant to change `entupleN` to `tupleN` as well. What do you think?

**Comment by jdegoes** [comment]
Oh, I see. The reason I choice entuple is because entuple is a verb (a made up one), while tuple is a noun. However, after thinking about it more, I like `tuple` naming convention because it matches `Tuple`, except for case.

**Comment by jdegoes** [comment]
Done.

**Comment by paf31** [comment]
Thanks!

**Pull request merged**